lightning/src/ln/channel.rs

```diff
@@ -4020,6 +4013,15 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 	}
 }
 
+fn is_unsupported_shutdown_script(their_features: &InitFeatures, scriptpubkey: &Script, is_outbound: bool) -> bool {
+	// BOLT 2 says we must only send a scriptpubkey of certain standard forms,
+	// which for a a BIP-141-compliant witness program is at max 42 bytes in length.
+	// So don't let the remote peer feed us some super fee-heavy script.
```

Comment:
We restrain shutdown scripts for at least different reasons :
a) to avoid fee-heavy script costing us a lot if we're funder (fees are only paid by funder for now)
b) to avoid transactions not propagating on the p2p tx-relay network (e.g sending you a 20-of-20 CHECKMULTISIG scriptpubkey, too big)

I think the first one is clear enough but would be better to underscore b), _standardness_ is a more restrictive notion than _smallness_

Reply:
Hmmm, this comment was already in place before this PR (see [here](https://github.com/rust-bitcoin/rust-lightning/blob/main/lightning/src/ln/channel.rs#L2908)). All I have done here is move it as a result of moving the validation code, and add a very small modification that you mentioned in this PR before.

If you want something different, can you be specific on the exact comment you want in the code? I can copy/paste what you said 👆 but not sure that's exactly what you want.

Reply:
I think the risk of non-standard transactions broadcasts should be mentioned, it's the reason why those checks should be applied independently of funder/fundee role.

"We restrain shutdown scripts to standards forms to avoid transactions not propagating on the p2p tx-relay network"